line_profiler/_line_profiler.pyx

```diff
@@ -308,6 +311,11 @@ cdef class LineProfiler:
         def __set__(self, value):
             self.threaddata.enable_count = value
 
+    # This is shared between instances, but thread-local
+    property _active_instances:
+        def __get__(self):
+            return self._active_instances_getter(threading.get_ident(), set())
```

Comment:
This allocates memory for a new set every time it is called even if that set isn't used (because the default item already exists). I'd opt for a try / except KeyError here, as the happy path in a try-except should be very low over head in recent python versions.